python_modules/dagster/dagster/_utils/error.py

```diff
@@ -81,11 +89,47 @@ def _serializable_error_info_from_tb(tb: traceback.TracebackException) -> Serial
 ]
 
 
+def _should_redact_user_code_error() -> bool:
+    return os.getenv("DAGSTER_REDACT_USER_CODE_ERRORS") == "1"
+
+
+def _get_masked_traceback_exception(with_instruction: bool) -> traceback.TracebackException:
+    """Raises a DagsterRedactedUserCodeError and returns the traceback exception for it.
+    Used to replace the traceback of a user code error with a generic message that they
+    can search for in their own logs.
+    """
+    error_id = str(uuid.uuid4())
+    try:
+        raise DagsterRedactedUserCodeError(
+            f"Error occurred during user code execution, error ID {error_id}"
+            + ("\nSearch in logs for this error ID for more details" if with_instruction else "")
+        )
+    except DagsterRedactedUserCodeError:
+        # Get the stack trace from the masking exception we just raised, instead of
+        # the underlying user code exception
+        exc_type, e, tb = sys.exc_info()
+        tb_exc = traceback.TracebackException(
+            check.not_none(exc_type), check.not_none(e), check.not_none(tb)
+        )
+        return tb_exc
```

Comment:
you can directly instantiate a `SerializableErrorInfo` only `message` is required. I don't think there is much utility to preserving a stack trace that doesn't point to anything useful.

Reply:
Cleaned up - removed `_get_masked_traceback_exception` in favor of directly instantiating the serializable info.